examples/common.go

```diff
@@ -46,3 +54,68 @@ func RedirectStdout(f func()) {
 
 	os.Stdout = old
 }
+
+func GetMapMetadata(name string, crc uint32) (metadata.Map, error) {
+	url := fmt.Sprintf("https://radar-overviews.csgo.saiko.tech/%s/%d/info.json", name, crc)
+
+	resp, err := http.Get(url)
+	if err != nil {
+		return metadata.Map{}, fmt.Errorf("failed to get map info.json from %q: %v", url, err)
+	}
+
+	defer resp.Body.Close()
+
+	var data map[string]interface{}
+
+	err = json.NewDecoder(resp.Body).Decode(&data)
+	if err != nil {
+		return metadata.Map{}, fmt.Errorf("failed to decode as JSON: %v", err)
+	}
+
+	mapInfo, ok := data[name].(map[string]interface{})
+	if !ok {
+		return metadata.Map{}, fmt.Errorf("failed to get map info.json entry for %q", name)
+	}
+
+	x, err := strconv.ParseFloat(mapInfo["pos_x"].(string), 64)
+	if err != nil {
+		return metadata.Map{}, fmt.Errorf("failed to get origin for X coordinate: %v", err)
+	}
+
+	y, err := strconv.ParseFloat(mapInfo["pos_y"].(string), 64)
+	if err != nil {
+		return metadata.Map{}, fmt.Errorf("failed to get origin for Y coordinate: %v", err)
+	}
+
+	scale, err := strconv.ParseFloat(mapInfo["scale"].(string), 64)
+	if err != nil {
+		return metadata.Map{}, fmt.Errorf("failed to get scale: %v", err)
+	}
```

Comment:
Looks to me like it would make sense to declare the type of this upfront and let the JSON parser handle heavy lifting?

E.g.
```go
type mapInfo struct {
    PosX float32 `json:"pos_x"`
    PosY float32 `json:"pos_y"`
    Scale float32 `json:"scale"`
}
```

Reply:
yeah, but the root looks like

```json
{
 "de_dust2": {...}
}
```

e.g.
- VDF: https://radar-overviews.csgo.saiko.tech/cs_agency/2230463619/info.txt
- JSON: https://radar-overviews.csgo.saiko.tech/cs_agency/2230463619/info.json

we want to keep that data as true to the original as possible, so getting rid of that root element /unesting is not really an option imo

We can probably 
- A: do some reflection foo-bar 
- or B: encode and re-decode the `de_dust2` part

to make this work properly, but that's not too nice either :cry: 

do you know a nice way of doing this?

Reply:
> EDIT: ha, alright, now I managed to find the format and realized why you're doing it like this. Sorry :)

no worries :smile: - certainly looks like a code smell at first glance .. wish go was better at handling arbitrary data sometimes :sweat:

Reply:
actually, `map[string]mapInfo` should work as type

https://stackoverflow.com/questions/18412126/golang-parse-a-json-with-dynamic-key

thanks for making me look at this again :smile:

Reply:
Nice! Yeah, that's quite a bit better :)